devlib/target.py

```diff
@@ -2892,6 +2895,99 @@ def _resolve_paths(self):
         if self.executables_directory is None:
             self.executables_directory = '/tmp/devlib-target/bin'
 
+class QEMULinuxTarget(LinuxTarget):
+    '''
+    Class for launching Linux target on QEMU.
+    '''
+
+    # pylint: disable=too-many-locals,too-many-arguments
+    def __init__(self,
+                 kernel_image,
+                 connection_settings=None,
+                 platform=None,
+                 working_directory=None,
+                 executables_directory=None,
+                 connect=True,
+                 modules=None,
+                 load_default_modules=True,
+                 shell_prompt=DEFAULT_SHELL_PROMPT,
+                 max_async=50,
+                 arch='aarch64',
+                 cpu_types='-cpu cortex-a72',
+                 initrd_image=str(),
+                 mem_size=512,
+                 num_cores=2,
+                 num_threads=2,
+                 cmdline='console=ttyAMA0',
+                 enable_kvm=True,
+                 boot_timeout=60,
+                 ):
+        super().__init__(connection_settings=connection_settings,
+                         platform=platform,
+                         working_directory=working_directory,
+                         executables_directory=executables_directory,
+                         connect=False,
+                         modules=modules,
+                         load_default_modules=load_default_modules,
+                         shell_prompt=shell_prompt,
+                         conn_cls=SshConnection,
+                         max_async=max_async)
+
+        qemu_executable = f'qemu-system-{arch}'
+        qemu_path = shutil.which(qemu_executable)
+        if qemu_path is None:
+            raise FileNotFoundError(f'Cannot find {qemu_executable} executable!')
+
+        if not os.path.exists(kernel_image):
+            raise FileNotFoundError(f'{kernel_image} does not exist!')
+
+        qemu_cmd = f'sudo {qemu_path} -kernel {kernel_image} -append "{cmdline}" ' \
```

Comment:
Good catch! Will fix it.

Reply:
`shlex.quote()` converts CPU list to `'-cpu cortex-a72 -cpu cortex-a53'` and QEMU is unhappy with those `'`.

Reply:
How did you use it ? You need to apply it to each parameter individually, otherwise it will add escaping so that the entire string is a single argument. That's partly why it's not recommended to use `shell=True` in `subprocess`.  It's typically a much better idea to simply pass an iterable, which won't require any escaping at all.